### ISSUE TITLE
Game crashes when pressing random in alts

### ISSUE THREAD

**Issue by ncorliss** [ImpactDevelopment/ImpactClient#1225]
When logged into an and then pressing random with only one alt in the list the game hard locks.

1. With 1 alt logged in
2. Log into that alt
3. Press random
4. Crash

**Expected behavior**
Nothing I guess. Pressed it by mistake.

<details> 
  <summary><strong>Game output</strong></summary>
 

```
[17:18:58] [main/INFO]: Loading tweak class name clientapi.load.ClientTweaker
[17:18:58] [main/INFO]: Using primary tweak class name clientapi.load.ClientTweaker
[17:18:58] [main/INFO]: Loading tweak class name baritone.launch.BaritoneTweaker
[17:18:58] [main/INFO]: Calling tweak class clientapi.load.ClientTweaker
[17:18:58] [main/INFO]: Injecting into ClassLoader
[17:18:58] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.7.10 Source=file:/home/mainlaptop/.minecraft/libraries/org/spongepowered/mixin/0.7.11-SNAPSHOT/mixin-0.7.11-SNAPSHOT.jar Service=LaunchWrapper Env=UNKNOWN
[17:18:58] [main/INFO]: FML platform manager could not load class cpw.mods.fml.relauncher.CoreModManager. Proceeding without FML support.
[17:18:58] [main/INFO]: Initialized Mixin bootstrap
[17:18:58] [main/INFO]: Setup Mixin Environment
[17:18:58] [main/INFO]: Registered Bytecode Transformes
[17:18:58] [main/INFO]: Compatibility level set to JAVA_8
[17:18:58] [main/ERROR]: Mixin config mixins.impact.json does not specify "minVersion" property
[17:18:58] [main/INFO]: Loaded Mixin Configurations
[17:18:58] [main/INFO]: Calling tweak class baritone.launch.BaritoneTweaker
[17:18:58] [main/ERROR]: Mixin config mixins.baritone.json does not specify "minVersion" property
[17:18:58] [main/INFO]: Loading tweak class name org.spongepowered.asm.mixin.EnvironmentStateTweaker
[17:18:58] [main/INFO]: Calling tweak class org.spongepowered.asm.mixin.EnvironmentStateTweaker
[17:18:58] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinBlock, name: clientapi.load.mixin.MixinBlock
[17:18:58] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinEntity, name: clientapi.load.mixin.MixinEntity
[17:18:58] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinEntityLivingBase, name: clientapi.load.mixin.MixinEntityLivingBase
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinEntityPlayerSP, name: clientapi.load.mixin.MixinEntityPlayerSP
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinEntityRenderer, name: clientapi.load.mixin.MixinEntityRenderer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinFontRenderer, name: clientapi.load.mixin.MixinFontRenderer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinGuiBossOverlay, name: clientapi.load.mixin.MixinGuiBossOverlay
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinGuiIngame, name: clientapi.load.mixin.MixinGuiIngame
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinGuiSpectator, name: clientapi.load.mixin.MixinGuiSpectator
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinItemRenderer, name: clientapi.load.mixin.MixinItemRenderer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinLayerArmorBase, name: clientapi.load.mixin.MixinLayerArmorBase
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinMinecraft, name: clientapi.load.mixin.MixinMinecraft
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinNetHandlerLoginClient, name: clientapi.load.mixin.MixinNetHandlerLoginClient
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinNetHandlerPlayClient, name: clientapi.load.mixin.MixinNetHandlerPlayClient
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinNettyPacketDecoder, name: clientapi.load.mixin.MixinNettyPacketDecoder
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinNettyPacketEncoder, name: clientapi.load.mixin.MixinNettyPacketEncoder
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinNetworkManager, name: clientapi.load.mixin.MixinNetworkManager
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinParticleManager, name: clientapi.load.mixin.MixinParticleManager
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinRender, name: clientapi.load.mixin.MixinRender
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinRenderItem, name: clientapi.load.mixin.MixinRenderItem
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinRenderLivingBase, name: clientapi.load.mixin.MixinRenderLivingBase
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinRenderManager, name: clientapi.load.mixin.MixinRenderManager
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.capi.json:MixinTimer, name: clientapi.load.mixin.MixinTimer
[17:18:59] [main/WARN]: Error loading class: Config (java.lang.ClassNotFoundException: The specified class 'Config' was not found)
[17:18:59] [main/WARN]: @Mixin target Config was not found mixins.impact.json:optifine.MixinConfig
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinAnvilChunkLoader, name: baritone.launch.mixins.MixinAnvilChunkLoader
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinBlockPos, name: baritone.launch.mixins.MixinBlockPos
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinChunkProviderClient, name: baritone.launch.mixins.MixinChunkProviderClient
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinChunkProviderServer, name: baritone.launch.mixins.MixinChunkProviderServer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinChunkRenderContainer, name: baritone.launch.mixins.MixinChunkRenderContainer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinChunkRenderWorker, name: baritone.launch.mixins.MixinChunkRenderWorker
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinEntityLivingBase, name: baritone.launch.mixins.MixinEntityLivingBase
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinEntityPlayerSP, name: baritone.launch.mixins.MixinEntityPlayerSP
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinEntityRenderer, name: baritone.launch.mixins.MixinEntityRenderer
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinMinecraft, name: baritone.launch.mixins.MixinMinecraft
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinNetHandlerPlayClient, name: baritone.launch.mixins.MixinNetHandlerPlayClient
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinNetworkManager, name: baritone.launch.mixins.MixinNetworkManager
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinRenderChunk, name: baritone.launch.mixins.MixinRenderChunk
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinRenderList, name: baritone.launch.mixins.MixinRenderList
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinVboRenderList, name: baritone.launch.mixins.MixinVboRenderList
[17:18:59] [main/ERROR]: Classloader restrictions [PACKAGE_CLASSLOADER_EXCLUSION] encountered loading mixins.baritone.json:MixinWorldClient, name: baritone.launch.mixins.MixinWorldClient
[17:18:59] [main/INFO]: Launching wrapped minecraft {net.minecraft.client.main.Main}
[17:18:59] [main/WARN]: @Inject(@At("INVOKE")) Shift.BY=2 on mixins.baritone.json:MixinEntityPlayerSP::handler$onPostUpdate$zed001 exceeds the maximum allowed value 0. Increase the value of maxShiftBy to suppress this warning.
[17:18:59] [main/INFO]: Setting user: MatrixGrenade
[17:19:01] [main/WARN]: Skipping bad option: lastServer:
[17:19:01] [main/INFO]: LWJGL Version: 2.9.4
[17:19:01] [main/INFO]: Reloading ResourceManager: Default, Faithful+1.12.2-rv4.zip
[17:19:01] [Sound Library Loader/INFO]: Starting up SoundSystem...
[17:19:01] [Thread-3/INFO]: Initializing LWJGL OpenAL
[17:19:01] [Thread-3/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[17:19:01] [Thread-3/INFO]: OpenAL initialized.
[17:19:02] [Sound Library Loader/INFO]: Sound engine started
[17:19:02] [main/INFO]: Created: 2048x1024 textures-atlas
[17:19:03] [main/INFO]: Narrator library successfully loaded
[17:19:04] [main/INFO]: Loading Fonts
[17:19:04] [main/INFO]: Checking for Updates
[17:19:04] [main/INFO]: Loading Users
[17:19:05] [main/INFO]: Initialized 115 mods!
[17:19:05] [main/FATAL]: Unable to load spam config file: Spam config file is in invalid format
[17:19:05] [main/INFO]: [CHAT] [Impact] Added FatalCreation as a friend!
[17:19:05] [main/INFO]: [CHAT] [Impact] Added Aashie as a friend!
[17:22:25] [pool-6-thread-1/INFO]: Logging in with username & password

```

</details>

<details> 
  <summary><strong>Crash report</strong></summary>
 

```
No crash report
```

</details>

<details> 
  <summary><strong>Additional Client Info</strong></summary>
 

- Impact Version:  4.6 for 1.12.2
- Using Optifine?  No
- Using LiteLoader?  No
- Enabled mods:  Jesus, Lev Control, No Slow, Riding, Sprint, Anti-Desync, Auto-Reconnect, Inv, Lag Meter, Middle Click Firend, Rainbow, Unpack, ClickGUI, ESP, HUD, Light, Main Menu, Nametags, StorageESP, Tracers, Trajectories, Auto-Eat, Respawn, Anti-Hunger, Portals.
- Operating System:  Pop_OS!(Ubuntu)

</details>

**Additional context**

**Comment by leijurv** [comment]
Confirmed

**Comment by leijurv** [comment]
```
            // Get a random alt that isn't logged in
            Alt alt;
            do alt = altManager.get(MathUtils.random(0, altManager.size() - 1));
            while (alt.getStatus() == Alt.Status.LOGGED_IN);
            login(alt);
```


this is due to 🦍 code written by brady.

if all your alts are logged in and you hit random this will OBVIOUSLY loop infinitely lol

**Comment by leijurv** [comment]
Fixed in 4.7
